src/SafeAreaView.tsx

```diff
@@ -1,22 +1,38 @@
 import * as React from 'react';
 import type {
+  EdgeRecord,
   NativeSafeAreaViewInstance,
   NativeSafeAreaViewProps,
 } from './SafeArea.types';
 import NativeSafeAreaView from './specs/NativeSafeAreaView';
+import { useMemo } from 'react';
 
 export type SafeAreaViewProps = NativeSafeAreaViewProps;
 
 export const SafeAreaView = React.forwardRef<
   NativeSafeAreaViewInstance,
   SafeAreaViewProps
 >(({ edges, ...props }, ref) => {
+  const nativeEdges = useMemo(() => {
+    return Array.isArray(edges)
+      ? edges.reduce<EdgeRecord>((accum, edge) => {
+          accum[edge] = 'additive';
+          return accum;
+        }, {})
+      : edges;
+  }, [edges]);
+
   return (
     <NativeSafeAreaView
       {...props}
-      // Codegen doesn't support default values for array types so
-      // set it here.
-      edges={edges ?? ['bottom', 'left', 'right', 'top']}
+      edges={
+        nativeEdges ?? {
```

Comment:
Can we move this object into the useMemo - or even better, outside the component - so we keep stable references? I know it was that way before, but we have the chance to fix it